### PR TITLE
GH-14809: [Dev] Add created GitHub issues to issues@arrow.apache.org

### LINKED ISSUE #14809: [Dev] Add created GitHub issues to issues@arrow.apache.org
### Describe the enhancement requested

There is an issues@arrow.apache.org mailing list: https://lists.apache.org/list.html?issues@arrow.apache.org
This list contains newly created issues.
At the moment new issues created on GitHub are not notified there, we should fix that.

### Component(s)

Developer Tools

### PR DESCRIPTION
**Pull request by assignUser**
* Closes: #14809

### CODE REVIEW

File: .asf.yaml [hunk 1]
```diff
@@ -25,6 +25,7 @@ github:
 
 notifications:
   commits:      commits@arrow.apache.org
+  issues_status: issues@arrow.apache.org
```
[the remark this answers is not in the document]
Comment (assignUser) [reply .asf.yaml:28]: That is a private repo only committers can see :( It looks like there is `issue` -> open/close and comments, `issues_comments` -> just comments and `issue_status` -> just open/close.

+1 on changing arrow-jl to _status

[the remark this answers is not in the document]
Comment (assignUser) [reply .asf.yaml:28]: It seems if you also add `issue_comments` it wil remove them from `issues`? https://github.com/apache/arrow-julia/blob/main/.asf.yaml#L23

[the remark this answers is not in the document]
Comment (kou) [reply .asf.yaml:28]: If a PR that changes https://github.com/apache/arrow-julia/blob/main/.asf.yaml#L22 to `issues_status`, I'll merge it.

**Review by kou — APPROVED** [review]
+1

File: .asf.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (raulcd) [reply .asf.yaml:28]: I've created the PR on the `arrow-julia` repository https://github.com/apache/arrow-julia/pull/366

**Comment by jorisvandenbossche** [comment]
In addition to julia, should we do the same for the other `arrow-..` repos? i.e. arrow-adbc, arrow-nanoarrow (and maybe also the different rust repos)

**Comment by kou** [comment]
I think so.

**Pull request merged**